kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/write/storageApi/StorageWriteApiDefaultStreamTest.java

```diff
@@ -0,0 +1,271 @@
+package com.wepay.kafka.connect.bigquery.write.storageApi;
+
+import com.google.api.core.ApiFuture;
+import com.google.cloud.bigquery.storage.v1.JsonStreamWriter;
+import com.google.cloud.bigquery.storage.v1.AppendRowsResponse;
+import com.google.cloud.bigquery.storage.v1.RowError;
+import com.google.cloud.bigquery.storage.v1.TableName;
+import com.google.cloud.bigquery.storage.v1.Exceptions;
+
+import com.google.rpc.Status;
+import com.wepay.kafka.connect.bigquery.exception.BigQueryStorageWriteApiConnectException;
+import io.grpc.StatusRuntimeException;
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.json.JSONObject;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.ArgumentMatchers;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.CALLS_REAL_METHODS;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.doNothing;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+public class StorageWriteApiDefaultStreamTest {
+
+    TableName mockedTableName = TableName.of("dummyProject", "dummyDataset", "dummyTable");
+
+    @Test
+    public void testDefaultStreamNoExceptions() throws Exception {
+        JsonStreamWriter mockedStreamWriter = mock(JsonStreamWriter.class);
+        SinkRecord mockedSinkRecord = mock(SinkRecord.class);
+        ApiFuture<AppendRowsResponse> mockedResponse = mock(ApiFuture.class);
+        AppendRowsResponse successResponse = AppendRowsResponse.newBuilder()
+                .setAppendResult(AppendRowsResponse.AppendResult.newBuilder().getDefaultInstanceForType()).build();
+        List<Object[]> testRows = new ArrayList<>();
+
+        testRows.add(new Object[]{mockedSinkRecord, new JSONObject()});
+
+        StorageWriteApiDefaultStream defaultStream = mock(StorageWriteApiDefaultStream.class, CALLS_REAL_METHODS);
+
+        doReturn(mockedStreamWriter).when(defaultStream).getDefaultStream(ArgumentMatchers.any());
+
+        when(mockedStreamWriter.append(ArgumentMatchers.any())).thenReturn(mockedResponse);
+        when(mockedResponse.get()).thenReturn(successResponse);
+
+        defaultStream.appendRows(mockedTableName, testRows, null);
+
+    }
+
+    @Test(expected = BigQueryStorageWriteApiConnectException.class)
+    public void testDefaultStreamNonRetriableError() throws Exception {
+        JsonStreamWriter mockedStreamWriter = mock(JsonStreamWriter.class);
+        SinkRecord mockedSinkRecord = mock(SinkRecord.class);
+        ApiFuture<AppendRowsResponse> mockedResponse = mock(ApiFuture.class);
```

Comment:
I see this code segment is common across all the test. Lets try to avoid duplication by having some helper function or initialising globally.

Reply:
Thanks for the inputs. Have updated the file.